tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
+        {
+            services.RemoveAll<IOpenIdConnectMiddlewareDiagnostics>();
+            services.AddSingleton<IOpenIdConnectMiddlewareDiagnostics>((provider) => diagnosticsMock);
+
+            var provider = services.BuildServiceProvider();
+
+            var oidcOptions = provider.GetRequiredService<IOptionsFactory<OpenIdConnectOptions>>().Create(_oidcScheme);
+
+            // Assert subscribed to diagnostics
+            if (subscribeToDiagnostics)
+                diagnosticsMock.ReceivedWithAnyArgs().Subscribe(Arg.Any<OpenIdConnectEvents>());
+            else
+                diagnosticsMock.DidNotReceiveWithAnyArgs().Subscribe(Arg.Any<OpenIdConnectEvents>());
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task AddSignIn_WithConfigName_B2cSpecificSetup(bool useServiceCollectionExtension)
+        {
+            var configMock = Substitute.For<IConfiguration>();
+            _configSection = GetConfigSection(_configSectionName, true);
+            configMock.Configure().GetSection(_configSectionName).Returns(_configSection);
+
+            var services = new ServiceCollection();
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(configMock, _configSectionName, _oidcScheme, _cookieScheme, false);
+            else
+                new AuthenticationBuilder(services)
+                    .AddSignIn(configMock, _configSectionName, _oidcScheme, _cookieScheme, false);
+
+            await AddSignIn_TestB2cSpecificSetup(services).ConfigureAwait(false);
+        }
+
         [Theory]
-        [InlineData(TestConstants.AuthorityCommonTenant, TestConstants.AuthorityCommonTenantWithV2)]
-        [InlineData(TestConstants.AuthorityOrganizationsUSTenant, TestConstants.AuthorityOrganizationsUSWithV2)]
-        [InlineData(TestConstants.AuthorityCommonTenantWithV2, TestConstants.AuthorityCommonTenantWithV2)]
-        [InlineData(TestConstants.AuthorityCommonTenantWithV2 + "/", TestConstants.AuthorityCommonTenantWithV2)]
-        [InlineData(TestConstants.B2CAuthorityWithV2, TestConstants.B2CAuthorityWithV2)]
-        [InlineData(TestConstants.B2CCustomDomainAuthorityWithV2, TestConstants.B2CCustomDomainAuthorityWithV2)]
-        [InlineData(TestConstants.B2CAuthority, TestConstants.B2CAuthorityWithV2)]
-        [InlineData(TestConstants.B2CCustomDomainAuthority, TestConstants.B2CCustomDomainAuthorityWithV2)]
-        public void EnsureAuthorityIsV2_0(string initialAuthority, string expectedAuthority)
-        {
-            OpenIdConnectOptions options = new OpenIdConnectOptions
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task AddSignIn_WithConfigActions_B2cSpecificSetup(bool useServiceCollectionExtension)
+        {
+            _configureMsOptions = (options) => {
+                options.Instance = TestConstants.AadInstance;
```

Comment:
B2C shouldn't use an AadInstance...use something like `"[https://fabrikamb2c.b2clogin.com](https://fabrikamb2c.b2clogin.com)"`

Reply:
Specific value doesn't affect the outcome of the test but I've updated it for readability.